### DISCUSSION TITLE
Laravel Octane on Windows

### DISCUSSION THREAD

**Discussion by schonhoff** [beyondcode/herd-community#1101] [Q&A]
Hey all,

is there a way to get Laravel Octane working on Windows with Laravel Herd? I tried some things (Roadrunner) but it won't work out. Maybe someone can help me out with it?

**Comment by ismaildasci** [comment]
Hey! 👋

Running **Laravel Octane** on Windows, especially with **Laravel Herd**, can be tricky because Octane is designed to work best in Unix-like environments. Here are a few options you can try to get it running on Windows:

### 1. Use WSL 2 (Windows Subsystem for Linux)
If you're not using WSL 2 yet, this might be the easiest approach. WSL 2 allows you to run a Linux distribution on Windows, providing better compatibility with tools like Roadrunner and Swoole, which are required by Octane.

- **Install WSL 2**: Follow the [Microsoft documentation](https://docs.microsoft.com/en-us/windows/wsl/install) to install WSL 2 with a Linux distribution (like Ubuntu).
- **Set up Laravel Octane in WSL 2**: Once WSL 2 is installed, install PHP, Node.js, and Composer in the Linux environment. Then, install Laravel Octane with Roadrunner or Swoole.

This approach lets you use Octane on Windows without needing a VM, and it closely resembles a native Linux setup.

### 2. Use Docker
Another option is to use Docker Desktop on Windows, which can run Octane through a Docker container. Laravel Herd doesn’t currently support Docker directly, but you can set up a separate Docker environment for your Octane project.

- **Create a Docker setup** with a PHP, Nginx, and MySQL stack. Add Roadrunner or Swoole to the Docker container.
- You can find some community Docker images that already include Laravel Octane with Roadrunner or Swoole, which might simplify setup.

### 3. Virtual Machine (VM) as a Last Resort
If WSL 2 and Docker don’t work for your setup, you could create a VM running Linux and install Laravel and Octane there. This approach is heavier but can work as a fallback if other methods aren’t feasible.

### Why Octane Doesn’t Work Natively on Windows
Octane uses high-performance servers like Swoole and Roadrunner, which are optimized for Unix-like systems. Windows compatibility is limited, so using an environment like WSL 2 or Docker helps bypass these limitations.

Let me know if this helps or if you run into any issues with these setups! 😊

**Reply by schonhoff** [reply]
Hey!

Thanks for the reply. If I understand you correctly there is no way to use octane on with Laravel Herd on Windows without using WSL or Docker? I tried out RoadRunner which has a Windows binary but it didn't work. Sadly Docker or WSL would add another stack to our developer process which we can't support/maintain currently. 

Would love to see something like FrankenPHP or Roadrunner to be added to Laravel Herd to support Octane out of the Box.

Cheers,
Michael
